packages/labs/context/src/test/context-provider_test.ts

```diff
@@ -225,3 +225,49 @@ memorySuite('memory leak test', () => {
     );
   });
 });
+
+test('regression test for https://github.com/lit/lit/issues/4158?', async () => {
+  const context = createContext<string>('my-context');
+
+  class Provider extends LitElement {
+    @provide({context})
+    name = 'b';
+
+    @provide({context})
+    n2 = 'a';
+
+    render() {
+      return html`<r-consumer></r-consumer><r-container></r-container>`;
+    }
+  }
+  customElements.define('r-provider', Provider);
+
+  class Consumer extends LitElement {
+    @consume({context, subscribe: true})
+    @property()
+    n1 = 'c';
+
+    render() {
+      return html`<p>${this.n1}</p>`;
+    }
+  }
+  customElements.define('r-consumer', Consumer);
+
+  class Container extends LitElement {
+    // uncomment for infinite loop:
+    @provide({context})
+    n2 = 'd';
+
+    render() {
+      return html`<r-consumer></r-consumer>`;
+    }
+  }
+  customElements.define('r-container', Container);
+
+  const provider = new Provider();
+  document.body.appendChild(provider);
+  await new Promise((r) => setTimeout(r, 100));
```

Comment:
what's the wait for? If you iloop, will it happen only after a render? I'd leave a comment on when the iloop will occur and see if you can tighten the wait here if possible, shouldn't a rAF or setTimeout(r, 0) do it?

Reply:
Tightened it to a setTimeout of 0